lib/_stream_writable.js

```diff
@@ -428,13 +428,11 @@ function onwriteError(stream, state, sync, er, cb) {
     // This can emit finish, and it will always happen
     // after error
     process.nextTick(finishMaybe, stream, state);
-    stream._writableState.errorEmitted = true;
     errorOrDestroy(stream, er);
   } else {
     // The caller expect this to happen before if
     // it is async
     cb(er);
-    stream._writableState.errorEmitted = true;
```

Comment:
I don’t think this change is considering all cases. There are a significant number of modules that override .destroy. Removing this line is going to break them.

I think this line should be retained, and a test added for that case.

Reply:
I put it back with some comments. I will need help with making such a test.

Reply:
@mcollina: Furthermore, I made the case you mention more consistent.

Reply:
@mcollina: actually this confuses me, if destroy is not overriden then this will actually cause the error not to be emitted and if it is overriden then errorEmitted isn't really relevant anymore? So I'm a little unsure where your concern is relevant?

Reply:
Also it is currently not set for the other error cases in this file?

Reply:
I think I need some example tests to fully understand the expected behaviour here. I suspect there might be bugs with the current implementation.

Reply:
Yes, that's very likely.

Reply:
My bad, I've read the code wrong.